### PR TITLE
Standardize on `wikilink`, `MediaWiki`, and `GitHub`

### PR DESCRIPTION
**Pull request by movermeyer**
### What are you trying to accomplish?

While working on https://github.com/foambubble/foam/pull/712, I noticed that we were using inconsistent terminology when it came to wikilinks and MediaWiki.

* [`wikilink`](https://en.wikipedia.org/wiki/Help:Link#Wikilinks) is the correct term, not `wiki-link`, nor `wiki link`. 
* [`MediaWiki`](https://www.mediawiki.org/wiki/MediaWiki) is the correct term, not `media-wiki`, nor `mediawiki`.
* [`GitHub`](https://github.com/) is the correct term, not `Github`.

Counterpart PR in `foam-template`: https://github.com/foambubble/foam-template/pull/39

### What approach did you choose and why?

A series of manual find-replace:

- `wiki[^l]link` is a good regex for this
- Renamed the wikilinks.md file and updated all of the link reference definitions

### What should reviewers focus on?

...

### The impact of these changes

Consistent terminology throughout Foam.

### Testing

...

### Checklist
- [ ] I have manually tested this change. 
- [x] This PR is safe to roll back.

### CODE REVIEW

**Comment by riccardoferretti** [comment]
I strongly agree with the spirit of the PR. 
This needs testing of the generated documentation to ensure it won't break links.
Not sure if you are familiar with creating the site locally (the closest documentation we have is in `publish-to-github-pages`), or let me know if you want me to run those tests for you.
Thanks!

**Comment by movermeyer** [comment]
> This needs testing of the generated documentation to ensure it won't break links.
@riccardoferretti I have now manually clicked through all the links using Jekyll and they all still work.

File: docs/.vscode/extensions.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
     // Tons of markdown goodies (lists, tables of content, so much more)
     "yzhang.markdown-all-in-one",
 
-    // [[wiki-links]], backlinking etc
+    // [[wikilinks]], backlinking etc
     "kortina.vscode-markdown-notes",
```
Comment (pderaaij) [inline docs/.vscode/extensions.json:18]: Not for this PR, but we should remove this extension from the repo as well.

Comment (movermeyer) [reply docs/.vscode/extensions.json:18]: https://github.com/foambubble/foam/pull/722

**Review by riccardoferretti — APPROVED** [review]
LGTM!

**Pull request merged**